shell.nix

```diff
@@ -1,3 +1,7 @@
+# To use, simly run `nix-shell` with the nix package manager installed. This
+# will drop you into a shell with all of the dependencies to run ./bin/qmk from
+# this repo.
```

Comment:
> In particular, I missed that the qmk binary is inside this repo at ./bin/qmk
rather than being provided by the upstream NixPkgs packaging of qmk_cli

Thats not quite true, `bin/qmk` is now considered deprecated and to be removed shortly. Instead users are to have `qmk_cli` installed globally.

```suggestion
# will drop you into a shell with all of the dependencies to run qmk subcommands
```